packages/react-core/src/demos/SearchInput.md

```diff
@@ -0,0 +1,299 @@
+      ref={searchInputRef}
+      id="custom-advanced-search"
+    />
+  );
+  
+  // Clicking the 'date within' toggle should open its associated menu and then place the browser 
+  // focus on the first menu item. 
+  const toggleDateWithinMenu = ev => {
+    ev.stopPropagation(); // Stop handleClickOutside from handling
+    setTimeout(() => {
+      if (dateWithinMenuRef.current) {
+        const firstElement = dateWithinMenuRef.current.querySelector('li > button:not(:disabled)');
+        firstElement && firstElement.focus();
+      }
+    }, 0);
+    setIsDateWithinOpen(!isDateWithinOpen);
+  };
+  
+  const dateWithinOptions = (
+    <Menu ref={dateWithinMenuRef} selected={dateWithin} onSelect={(e, itemId) => {
+      e.stopPropagation();
+      setIsDateWithinOpen(false);
+      setDateWithin(itemId);
+    }}>
+      <MenuContent>
+        <MenuList>
+          <MenuItem itemId="1 day">1 day</MenuItem>
+          <MenuItem itemId="3 days">3 days</MenuItem>
+          <MenuItem itemId="1 week">1 week</MenuItem>
+          <MenuItem itemId="2 weeks">2 weeks</MenuItem>
+          <MenuItem itemId="1 month">1 month</MenuItem>
+          <MenuItem itemId="2 months">2 months</MenuItem>
+          <MenuItem itemId="6 months">6 months</MenuItem>
+          <MenuItem itemId="1 year">1 year</MenuItem>
+        </MenuList>
+      </MenuContent>
+    </Menu>
+  );
+  
+  const dateWithinToggle = (
+    <MenuToggle ref={dateWithinToggleRef} onClick={toggleDateWithinMenu} isExpanded={isDateWithinOpen} style={{width: "100%"}}>
+      {dateWithin}
+    </MenuToggle>
+  );
+  
+  const advancedForm = (
+    <div ref={advancedSearchPaneRef} role="dialog" aria-label="Advanced search form">
+      <Form>
+        <Card>
+          <CardBody>
+            <Grid hasGutter md={6}>
+              <GridItem span={12}>
+                <FormGroup label='Has the words' fieldId='has-words' key='has-words'>
+                  <TextInput
+                    type='text'
+                    id='has-words'
+                    value={hasWords}
+                    onChange={value => {
+                      setHasWords(value);
+                      setValue(value);
+                    }}
+                    ref={firstAttrRef}
+                  />
+                </FormGroup>
+              </GridItem>
+              <FormGroup label='Date within' fieldId='date-within' key='date-within'>
+                <Popper trigger={dateWithinToggle} popper={dateWithinOptions} isVisible={isDateWithinOpen} />
+              </FormGroup>
+              <FormGroup label='Of date' fieldId='date' key='date'>
+                <DatePicker 
+                  id="datePicker" 
+                  style={{width: "100%"}} 
+                  value={date} 
+                  onChange={setDate} 
+                  appendTo={() => document.querySelector("#datePicker")}
+                />
+              </FormGroup>
+            </Grid>
+          </CardBody>
+          <CardFooter>
```

Comment:
The spacing for card footer actions doesn't match the spacing for form actions. If we swap the card out with the search input menu/menu-body elements, we can use a form action group for the right spacing here. If we aren't going to do that, I think this form should just be in a `<CardBody>`, and the form action group used for the actions.

Reply:
I can do that!

Reply:
well - actually.  I will play with it - I think i was getting into sticky territory where the actions need to be in a `Form` and so the `Form` was wrapping the `CardBody` and `CardFooter`. I'll take another stab at it